тз16

russuAV opened the pull request:
• реализовано создание запроса на добавление вещи и ответ на него
	• вынесена валидация входящих запросов в gateway
	• добавлены тесты

russuAV committed:
• реализовано создание запроса на добавление вещи и ответ на него
	• вынесена валидация входящих запросов в gateway
	• добавлены тесты

Sla-als commented on gateway/src/main/java/ru/practicum/booking/BookingClient.java:
```diff
@@ -0,0 +1,61 @@
+    private static final String API_PREFIX = "/bookings";
+
+
+    public BookingClient(@Value("${shareit-server.url}") String serverUrl, RestTemplateBuilder builder) {
```
Лучше переименовать свойство в application.properties и в @Value в shareit.server.url (с точкой).

Sla-als commented on common/src/main/java/ru/practicum/item/NewItemRequest.java:
- @NotBlank уже включает проверку на null, поэтому @NotNull избыточен.
-  нет аннотации @Future / @FutureOrPresent на даты, хотя вы проверяете только, что end > start - 
нужно добавить @FutureOrPresent/@Future на start/end, если хотите запретить прошлые даты.

Sla-als reviewed (commented):
Есть небольшие замечания

russuAV committed:
• удалены лишние @NotNull при использовании @NotBlank
	• переименовано свойство в application.properties и в @value на shareit.server.url
	• добавлена аннотация @future для проверки введенных пользователем дат бронирования
